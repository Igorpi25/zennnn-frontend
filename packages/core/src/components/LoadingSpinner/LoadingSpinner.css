.loading-spinner {
  @apply inline-flex align-middle;
}
.loading-spinner__wrapper {
  @apply inline-block relative align-middle;
}
.loading-spinner__ray {
  @apply block absolute bg-current;
}

@keyframes loading-spinner {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
