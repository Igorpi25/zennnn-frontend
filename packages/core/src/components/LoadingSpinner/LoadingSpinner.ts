import { h, FunctionalComponent } from 'vue'

const DEFAULT_SIZE = 18
const RAYS_COUNT = 12
const SPEED = 1.2

const LoadingSpinner: FunctionalComponent<{
  size: number | string,
}> = (props) => {
  const size = Number(props.size || DEFAULT_SIZE)
  const radius = size * 0.35
  const width = size * 0.08
  const height = size * 0.22
  const halfSize = Math.round(size / 2)

  const childStyle = {
    transformOrigin: `${halfSize}px ${halfSize}px`,
    animation: `loading-spinner ${SPEED}s linear infinite`,
  }

  const children = Array(RAYS_COUNT).fill(null).map((_, i) => {
    return h('div', {
      style: {
        ...childStyle,
        transform: `rotate(${i * 30}deg)`,
        animationDelay: `${SPEED * -1 * (RAYS_COUNT - 1 - i) / RAYS_COUNT}s`,
      },
    }, [h('div', {
      class: 'loading-spinner__ray',
      style: {
        borderRadius: '20%',
        top: `${Math.round(halfSize - radius - height / 2)}px`,
        left: `${Math.round(halfSize - width / 2)}px`,
        width: `${Math.round(width)}px`,
        height: `${Math.round(height)}px`,
      },
    })])
  })

  return h(
    'div',
    { class: 'loading-spinner' },
    [
      h('div', {
        class: 'loading-spinner__wrapper',
        style: { width: `${size}px`, height: `${size}px` },
      }, children),
    ],
  )
}

export default LoadingSpinner
