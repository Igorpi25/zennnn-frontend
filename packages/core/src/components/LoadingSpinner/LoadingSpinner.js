import { h } from 'vue'

import './LoadingSpinner.css'

/**
 * Props
 * @param {string} [props.size=18] The size of loader.
 */
export default (props) => {
  const DEFAULT_SIZE = 18
  const RAYS_COUNT = 12
  const speed = 1.2
  const size = props.size || DEFAULT_SIZE
  const radius = size * 0.35
  const width = size * 0.08
  const height = size * 0.22
  const halfSize = Math.round(size / 2)

  const childStyle = {
    transformOrigin: `${halfSize}px ${halfSize}px`,
    animation: `loading-spinner ${speed}s linear infinite`,
  }

  const children = Array(RAYS_COUNT).fill().map((_, i) => {
    return h('div', {
      style: {
        ...childStyle,
        transform: `rotate(${i * 30}deg)`,
        animationDelay: `${speed * -1 * (RAYS_COUNT - 1 - i) / RAYS_COUNT}s`,
      },
    }, [h('div', {
      class: 'loading-spinner__ray',
      style: {
        borderRadius: '20%',
        top: `${Math.round(halfSize - radius - height / 2)}px`,
        left: `${Math.round(halfSize - width / 2)}px`,
        width: `${Math.round(width)}px`,
        height: `${Math.round(height)}px`,
      },
    })])
  })

  return h(
    'div',
    { class: 'loading-spinner' },
    [
      h('div', {
        class: 'loading-spinner__wrapper',
        style: { width: `${size}px`, height: `${size}px` },
      }, children),
    ],
  )
}
