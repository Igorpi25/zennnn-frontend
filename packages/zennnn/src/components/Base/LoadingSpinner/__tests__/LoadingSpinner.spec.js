import { mount } from '@vue/test-utils'
import LoadingSpinner from '../LoadingSpinner'

describe('LoadingSpinner.js', () => {
  let mountFunction

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(LoadingSpinner, {
        ...options,
      })
    }
  })

  // Render

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with size prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        size: 24,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
