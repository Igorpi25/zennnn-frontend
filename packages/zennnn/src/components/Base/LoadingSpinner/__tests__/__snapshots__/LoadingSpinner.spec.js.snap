// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner.js should render component and match snapshot 1`] = `
<div class="loading-spinner">
  <div class="loading-spinner__wrapper" style="width: 18px; height: 18px;">
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(0deg); animation-delay: -1.0999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(30deg); animation-delay: -1s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(60deg); animation-delay: -0.8999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(90deg); animation-delay: -0.7999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(120deg); animation-delay: -0.7000000000000001s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(150deg); animation-delay: -0.6s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(180deg); animation-delay: -0.5s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(210deg); animation-delay: -0.39999999999999997s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(240deg); animation-delay: -0.3s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(270deg); animation-delay: -0.19999999999999998s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(300deg); animation-delay: -0.09999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
    <div style="transform-origin: 9px 9px; animation: loading-spinner 1.2s linear infinite; transform: rotate(330deg); animation-delay: 0s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 8px; width: 1px; height: 4px;"></div>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner.js should render component with size prop and match snapshot 1`] = `
<div class="loading-spinner">
  <div class="loading-spinner__wrapper" style="width: 24px; height: 24px;">
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(0deg); animation-delay: -1.0999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(30deg); animation-delay: -1s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(60deg); animation-delay: -0.8999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(90deg); animation-delay: -0.7999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(120deg); animation-delay: -0.7000000000000001s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(150deg); animation-delay: -0.6s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(180deg); animation-delay: -0.5s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(210deg); animation-delay: -0.39999999999999997s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(240deg); animation-delay: -0.3s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(270deg); animation-delay: -0.19999999999999998s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(300deg); animation-delay: -0.09999999999999999s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
    <div style="transform-origin: 12px 12px; animation: loading-spinner 1.2s linear infinite; transform: rotate(330deg); animation-delay: 0s;">
      <div class="loading-spinner__ray" style="border-radius: 20%; top: 1px; left: 11px; width: 2px; height: 5px;"></div>
    </div>
  </div>
</div>
`;
